src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -159,30 +159,14 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 	char *colocateWithTableName = NULL;
 	bool viaDeprecatedAPI = true;
 
-	/*
-	 * Lock target relation with an exclusive lock - there's no way to make
-	 * sense of this table until we've committed, and we don't want multiple
-	 * backends manipulating this relation.
-	 */
-	Relation relation = try_relation_open(relationId, ExclusiveLock);
```

Comment:
we may need to keep this lock (unless we also take it in CreateDistributedTable)

Reply:
We call `BuildDistributionKeyFromColumnName` in `CreateDistributedTable`, and we acquire the lock in that func

Reply:
That seems risky since this is an aggressive lock that blocks writes. I don't know if all the code paths that call BuildDistributionKeyFromColumnName  intend to block writes.

Reply:
Discussed offline, it was probably not quite correct to take the lock here because we might drop the table if we need an undistribute step.